application/app/src/main/java/com/example/buber/Views/Activities/MapActivity.java

```diff
@@ -99,66 +111,97 @@ protected void onCreate(Bundle savedInstanceState) {
         // INSTANTIATE RIDER MAIN ACTION BUTTONS
         riderRequestMainBtn = findViewById(R.id.rider_request_btn);
         riderRequestCancelMainBtn = findViewById(R.id.rider_request_cancel_btn);
+        riderCancelPickupBtn = findViewById(R.id.rider_cancel_pickup_btn);
 
         // INSTANTIATE DRIVER MAIN ACTION BUTTONS
         driverShowRequestsMainBtn = findViewById(R.id.driver_show_requests_btn);
 
         // INSTANTIATE STATUS BUTTON
         statusButton = findViewById(R.id.rideStatus);
-        statusButton.setEnabled(false);
 
         // HIDE SIDEBAR
         hideSettingsPanel();
 
-        App.getModel().addObserver(this);
-
+        // INSTANTIATE NOTIFICATION MANAGER
+        notificationManager = NotificationManagerCompat.from(this);
 
+        App.getModel().addObserver(this);
         Trip sessionTrip = App.getModel().getSessionTrip();
         if (sessionTrip != null) {
             currentTripStatus = sessionTrip.getStatus();
+            statusButton.setVisibility(View.VISIBLE);
             showActiveMainActionButton();
-            statusButton.setEnabled(true);
         }
     }
 
     /***** OBSERVING OBSERVABLES ******/
-    /**update method updates the activity when necessary
-     * @param o,arg are instances of the observable and object*/
+    /**
+     * update method updates the activity when necessary, USED MOSTLY FOR NOTIFICATIONS
+     *
+     * @param o,arg are instances of the observable and object
+     */
     @Override
     public void update(Observable o, Object arg) {
         Trip sessionTrip = App.getModel().getSessionTrip();
-        User sessionUser = App.getModel().getSessionUser();
+        User.TYPE currentUserType = App.getModel().getSessionUser().getType();
+
+        // Handles State Resets regarding Trip Cancellations
         if (sessionTrip == null) {
```

Comment:
Can we add a else clause that processes any unintended logic for error handling purposes.